requirements-base.txt

```diff
@@ -97,7 +97,7 @@ cryptography>=38.0.3
 # Note, grpcio>1.30.0 requires setting GRPC_POLL_STRATEGY=epoll1
 # See https://github.com/grpc/grpc/issues/23796 and
 # https://github.com/grpc/grpc/blob/v1.35.x/doc/core/grpc-polling-engines.md#polling-engine-implementations-in-grpc
-grpcio>=1.59.0
+grpcio>=1.66.1
```

Comment:
it's really unclear to me why we'd need to update this if it's unused?

what was broken that prompted this?

Reply:
We're trying to break this pr apart and merge it in piece by piece:
https://github.com/getsentry/sentry/pull/76394
Which requires this version of grpc.

It also helps us test our changes when we switch to that branch, because whenever we pull master, our grpc version get override to the version at master.

Reply:
> We're trying to break this pr apart and merge it in piece by piece:
> https://github.com/getsentry/sentry/pull/76394
> Which requires this version of grpc.

I understand that part -- but *why* does it require that version of grpcio?  that part is not at all clear

I'm especially suspicious as the TSC agreement was that grpc was not going to be used at runtime (only protobuf over http)

Reply:
There was another TSC meeting this week. The blockers around with regards to load-
balancing gRPC in production have been resolved.  There weren't any other objections raised during that session. 

I'd like to move forward with this as currently we can't make progress in local development with grpc. On import service stubs raise errors about incompatible grpcio libraries, which blocks us during CI from merging any code that uses grpc services, or grpc client stubs.